features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SessionReplayResource.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder
+
+import java.io.Serializable
+
+// An individual resource. It is used to describe binary representation of heavy resources such as images.
+internal data class SessionReplayResource(
+    var identifier: String,
+    var data: ByteArray,
+    var context: SessionReplayResourceContext
+) : Serializable {
```

Comment:
why does it need to be `Serializable`?

Reply:
I did this because we will need to serialize this object in order to insert it in the `RawBatchEvent` data field, and in order to do so it would need to be serializable